shell/platform/common/alert_platform_node_delegate.h

```diff
@@ -0,0 +1,41 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_COMMON_ALERT_PLATFORM_NODE_DELEGATE_H_
+#define FLUTTER_SHELL_PLATFORM_COMMON_ALERT_PLATFORM_NODE_DELEGATE_H_
+
+#include "flutter/third_party/accessibility/ax/ax_node_data.h"
+#include "flutter/third_party/accessibility/ax/platform/ax_platform_node_delegate_base.h"
+
+namespace flutter {
+
+class AlertPlatformNodeDelegate : public ui::AXPlatformNodeDelegateBase {
+ public:
+  explicit AlertPlatformNodeDelegate(
+      ui::AXPlatformNodeDelegate* parent_delegate);
+  ~AlertPlatformNodeDelegate();
+
+  AlertPlatformNodeDelegate(const AlertPlatformNodeDelegate& other) = delete;
+  AlertPlatformNodeDelegate operator=(const AlertPlatformNodeDelegate& other) =
+      delete;
+
+  void SetText(const std::u16string& text);
```

Comment:
I'm also leaning towards keeping this a string reference, but, the string reference is also copied when `AXNodeData` stores it as an attribute.

Reply:
> The AXNodeData methods that this method calls take string& parameters. Am I mistaken that constructing a string from a string_view effectively copies it?

Yes - to benefit from this, you'd need any methods this gets passed to to also take a view rather than a string ref. If the current implementation is doing so, then leave as-is; at some point we can do a sweep of the codebase to clean these up across the board.